tests/peepopt/muldiv_c.ys

```diff
@@ -0,0 +1,258 @@
+# Basic pattern transformed: (a * b) / c
+read_verilog <<EOT
+module top(
+	input signed [3:0] a,
+	output signed [7:0] y,
+);
+    wire signed [7:0] mul;
+	assign mul = a * 4'sd6;
+	assign y = mul / 8'sd3;
+endmodule
+EOT
+equiv_opt -assert peepopt
```

Comment:
@widlarizer please convert all equiv_opt to the miter/sat technique.
equiv_opt does not converge if internal nets are removed/replaced, where miter/sat does.
See example in the file in this PR:
tests/peepopt/multdiv_c.ys

Reply:
equiv_opt uses the same sat solver invocations under the hood so this shouldn't be correct. equiv_opt should be preferred over manual sat invocations wherever possible as it is much less prone to errors where the wrong arguments are selected (e.g. see #4629). You will see that if you use the miter based commands you suggest for these tests they still fail, so I suspect the peepopt is being applied incorrectly in those cases.

Reply:
I've ran into this symptom as well, so I've switched temporarily to some hack, and when I removed it again, there was no problem. This would match with what George is saying, I've been happy with `equiv_opt -assert`

Reply:
@georgerennie the problem with equiv_opt is that it tries to prove internal signals that don't exist anymore after the peepopt. The miter/sat invocation only operates on primary outputs and passes.
Your statement "You will see that if you use the miter based commands you suggest for these tests they still fail" is incorrect. I ran all the PRs I submitted for the test I edited. They pass. I even injected errors to make sure the miter/sat catches them and it did.
Can you provide a proof that any of the tests submitted actually pass with the equiv_opt? For me they all fail on internal signals. Is there any option for it to consider only the primary outputs? Else these transforms all change the internal structure too much to be proven.

Reply:
`equiv_opt` will match on internal nets, but only ones that exist and have a driver on both the pre and post optimized netlists. Optimization passes should not change the semantics of these nets (with the exception of refining undefined values) and im pretty sure these peephole obey that.

> Can you provide a proof that any of the tests submitted actually pass with the equiv_opt?

On https://github.com/georgerennie/yosys/tree/george/muldiv_c_equiv_opt I have taken the latest commit of this PR (1c774f4) and rewritten `multdiv_c.ys` to use `equiv_opt` instead of manual commands. This passes on my machine (and in less time, 19s vs 24s).

On that branch I have also commented out all the currently failing tests from `muldiv_c.ys`. All remaining tests pass using `equiv_opt`, and if you uncomment others you will see that they still fail using the miter/sat invocation, I suspect because the optimization is being applied incorrectly. I have put comments starting with `# GEORGE` in that file with some other notes.

Reply:
Ahh I see now that I missed the muxadd testcases. These do indeed fail with `equiv_opt`, but I believe that is a fault of the pass, e.g.
```verilog
wire [3:0] ab = a + b;
assign y = s ? ab : a;
```
is rewritten to something like
```verilog
wire [3:0] _0_ = a + (s ? ab : a);
wire [3:0] ab = _0_;
assign y = _0_;
```

This is not a valid rewrite application as it changes the meaning of `ab`, even though you only apply the rewrite in cases where `ab` has no other fanout. This is problematic for usecases like formal verification where suddenly an internal net shows incorrect values in a trace. To get around this you should probably just leave the original `ab = a + b` in the circuit. If we are doing synthesis it will be trivially cleaned up by `opt_clean` but it may be useful for other users if it remains.

Reply:
I think renaming the signal (to a new private name) is also fine when it no longer has the same meaning and we know there are no external fanouts. From a practical standpoint It's desirable for passes to be verifiable with equiv_opt because that's the go-to tool for tracking down correctness bugs over full designs and flows.

Reply:
Yeah this works too and I would be happy with that

Reply:
Fixed here: https://github.com/akashlevy/yosys/pull/15